File: tournaments/findPath/findPath.cs
bool findPath(int[][] matrix) {
    int n=matrix.Length;
    int m= matrix[0].Length;
    int a=0; int b=0;
    for (int i=0; i<n; i++)
        for (int j=0; j<m; j++)
            if (matrix[i][j]==1){
                a=i;
                b=j;
            }
    for (int k=2;k<=m*n;k++)
    {
        if (a>0 && matrix[a-1][b]==k){a--;continue; }
        if (b>0 && matrix[a][b-1]==k){b--;continue; }
        if (a<n-1 && matrix[a+1][b]==k){a++;continue; }
        if (b<m-1 && matrix[a][b+1]==k){b++;continue; }
        return false;
    }
    return true;
}

File: tournaments/findPath/findPath.js
function findPath(matrix) {
    const getNext = (i, j, item) => {
        for (const [dRow, dCol] of [[1, 0], [0, 1], [-1, 0], [0, -1]]) {
            if ((i + dRow) in matrix && (j + dCol) in matrix[0] && matrix[(i + dRow)][(j + dCol)] === 1 + item) {
                return [(i + dRow), (j + dCol)];
            }
        }

        throw new Error();
    };

    const getStart = () => {
        for (let i = 0; rows > i; ++i) {
            for (let j = 0; cols > j; ++j) {
                if (1 === matrix[i][j]) {
                    return [i, j];
                }
            }
        }

        throw new Error();
    };

    const rows = matrix.length;
    const cols = matrix[0].length;
    const count = rows * cols;
    let index = 1;
    let [row, col] = getStart();

    while (count > index) {
        try {
            [row, col] = getNext(row, col, index);
            ++index;
        } catch(e) {
            return false;
        }
    }

    return true;
}
